cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicatorBuilder.java

```diff
@@ -372,6 +379,19 @@ public E addResponseInterceptors(List<HttpConnectionResponseInterceptor> interce
         return (E) this;
     }
 
+    public E username(String username) {
+        this.username = username;
```

Comment:
I think it will help prevent potential NPEs down the line (e.g. when trying to encode in the `CookieInterceptor`). I can't think of a valid use case for passing `null` to the method (vs. say just omitting the call) and we are relying on `null` to tell us that username and password have not been set so we should probably "reserve" that state.